File: src/main/kotlin/no/nav/tag/innsynAareg/services/sts/StsCacheConfig.java
package no.nav.tag.innsynAareg.services.sts;

import com.github.benmanes.caffeine.cache.Caffeine;
import org.springframework.cache.caffeine.CaffeineCache;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.concurrent.TimeUnit;

@Configuration
public class StsCacheConfig {

    final static String STS_CACHE = "sts_cache";

    @Bean
    public CaffeineCache stsCache() {
        return new CaffeineCache(STS_CACHE,
                Caffeine.newBuilder()
                    .maximumSize(1)
                    .expireAfterWrite(59, TimeUnit.MINUTES)
                    .recordStats()
                    .build());
    }
}

File: src/main/kotlin/no/nav/tag/innsynAareg/client/sts/STSCacheConfig.kt
package no.nav.tag.innsynAareg.client.sts

import com.github.benmanes.caffeine.cache.Caffeine
import org.springframework.cache.caffeine.CaffeineCache
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.concurrent.TimeUnit

@Configuration
class STSCacheConfig {
    companion object {
        const val STS_CACHE = "sts_cache"
    }

    @Bean
    fun stsCache(): CaffeineCache {
        return CaffeineCache(
            STS_CACHE,
            Caffeine.newBuilder()
                .maximumSize(1)
                .expireAfterWrite(40, TimeUnit.MINUTES)
                .recordStats()
                .build()
        )
    }
}
